### DISCUSSION TITLE
simulation stop time

### DISCUSSION THREAD

**Discussion by piotre13** [GMLC-TDC/HELICS#2673] [Q&A]
I'm wondering:  if I do not set the stoptime, does it have a default value? I'm wondering this because i'm performing very long simulations with number of seconds > 9e9 seconds and I was not setting the stop time. around the value of the MAX_TIME property (Big number definition 9223372036.854774) i started experimenting time mismatches. can this be related to the fact that I'm simulating for a period longer than this value?

**Comment by trevorhardy** [comment]
MAX_TIME is defined like you stated and that works out to be 292 years; are you sure you really need to simulate longer than that? If you just need a large number of time steps, HELICS has time resolution down to 1ns so you can advance time at that size and get nine orders of magnitude more timesteps.

If you're saying that you're trying to simulate beyond MAX_TIME and you're getting time grant mis-matches I wouldn't be surprised. There are limits to the times that can be represented in HELICS but we're hoping 292 years worth of nanoseconds would be enough.

**Comment by trevorhardy** [comment]
To answer your question more directly, I don't think we have a default time at which HELICS will say, "enough", stop granting time, and terminate the processes. If you are trying to run the co-simulation beyond MAX_TIME and it's throwing warnings, maybe we should add that in.

**Comment by phlptp** [comment]
It is possible to recompile HELICS to change the ns resolution to something else.   Which would be required if you wanted to for example simulation the age of the universe or something like that, or needed atomic scale times in picoseconds.  nanoseconds seemed like a good balance for the 99.99% of existing use cases.   But the underlying time codes can be changed at compile time.   Or just by assuming in the applications a different scale.

**Comment by trevorhardy** [comment]
@piotre13; is this still an open issue for you?
